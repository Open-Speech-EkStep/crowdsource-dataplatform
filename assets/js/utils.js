const { HOUR_IN_SECONDS, SIXTY } = require("./constants");

const ALL_LANGUAGES = [
    {value: "Assamese",id: "as", text: "অসমীয়া", hasLocaleText: true},
    {value: "Bengali", id: "bn", text: "বাংলা", hasLocaleText: true},
    {value: "English", id: "en", text: "English", hasLocaleText: true},
    {value: "Gujarati", id: "gu", text: "ગુજરાતી", hasLocaleText: true},
    {value: "Hindi", id: "hi", text: "हिंदी", hasLocaleText: true},
    {value: "Kannada", id: "kn", text: "ಕನ್ನಡ", hasLocaleText: true},
    {value: "Malayalam", id: "ml", text: "മലയാളം", hasLocaleText: true},
    {value: "Marathi", id: "mr", text: "मराठी", hasLocaleText: true},
    {value: "Odia", id: "or", text: "ଓଡିଆ", hasLocaleText: true},
    {value: "Punjabi", id: "pa", text: "ਪੰਜਾਬੀ", hasLocaleText: true},
    {value: "Tamil", id: "ta", text: "தமிழ்", hasLocaleText: true},
    {value: "Telugu", id: "te", text: "తెలుగు", hasLocaleText: true}];

function convertPXToVH(px) {
    return px * (100 / document.documentElement.clientHeight);
}

function setPageContentHeight() {
    const $footer = $('footer');
    const $nav = $('.navbar');
    const edgeHeightInPixel = $footer.outerHeight() + $nav.outerHeight()
    const contentHeightInVH = 100 - convertPXToVH(edgeHeightInPixel)
    $('#content-wrapper').css('min-height', contentHeightInVH + 'vh');
}

function toggleFooterPosition() {
    const $footer = $('footer');
    $footer.toggleClass('fixed-bottom')
    $footer.toggleClass('bottom')
}

function fetchLocationInfo() {
    //https://api.ipify.org/?format=json
    return fetch('https://www.cloudflare.com/cdn-cgi/trace').then(res => res.text()).then(ipAddressText => {
        const dataArray = ipAddressText.split('\n');
        let ipAddress = "";
        for (let ind in dataArray) {
            if (dataArray[ind].startsWith("ip=")) {
                ipAddress = dataArray[ind].replace('ip=', '');
                break;
            }
        }
        if (ipAddress.length !== 0) {
            return fetch(`/location-info?ip=${ipAddress}`);
        } else {
            return new Promise((resolve, reject) => {
                reject("Ip Address not available")
            })
        }
    });
}

const performAPIRequest = (url) => {
    return fetch(url).then((data) => {
        if(!data.ok) {
            throw Error(data.statusText || 'HTTP error');
        } else {
            return Promise.resolve(data.json());
        }
    });
}

const getLocaleString = function() {
    performAPIRequest('/get-locale-strings')
        .then((response) => {
            localStorage.setItem('localeString', JSON.stringify(response));
        });
}

const updateLocaleLanguagesDropdown = (language) => {
    const dropDown = $('#localisation_dropdown');
    const localeLang = ALL_LANGUAGES.find(ele => ele.value === language);
    if (language.toLowerCase() === "english" || localeLang.hasLocaleText === false) {
        dropDown.html('<a id="english" class="dropdown-item" href="/changeLocale/en">English</a>');
    } else {
        dropDown.html(`<a id="english" class="dropdown-item" href="/changeLocale/en">English</a>
        <a id=${localeLang.value} class="dropdown-item" href="/changeLocale/${localeLang.id}">${localeLang.text}</a>`);
    }
}

const calculateTime = function (totalSeconds, isSeconds = true) {
    const hours = Math.floor(totalSeconds / HOUR_IN_SECONDS);
    const remainingAfterHours = totalSeconds % HOUR_IN_SECONDS;
    const minutes = Math.floor(remainingAfterHours / SIXTY);
    const seconds = Math.round(remainingAfterHours % SIXTY);
    if (isSeconds) {
        return { hours, minutes, seconds };
    } else {
        return { hours, minutes };
    }
};

const formatTime = function (hours, minutes = 0, seconds = 0) {
    let result = "";
    if (hours > 0) {
        result += `${hours} hrs `;
    }
    if (minutes > 0) {
        result += `${minutes} min `;
    }
    if (hours === 0 && minutes === 0 && seconds > 0) {
        result += `${seconds} sec `;
    }
    return result.substr(0, result.length - 1);
};

module.exports = { setPageContentHeight, toggleFooterPosition, fetchLocationInfo, updateLocaleLanguagesDropdown ,calculateTime, formatTime, getLocaleString, performAPIRequest}
