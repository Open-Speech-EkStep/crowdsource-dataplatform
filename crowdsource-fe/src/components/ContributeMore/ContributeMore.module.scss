.root {
  border-radius: 12px;
  box-shadow: 0 4px 16px 4px rgba(0, 123, 255, 0.2);
  max-width: 64rem;
}

.medal {
  @include media-breakpoint-up(md) {
    height: 3.25rem;
    width: 2.5rem;
  }
}

.upcomingBadge {
  opacity: 0.8;
}

.otherBadges {
  opacity: 0.4;
}
