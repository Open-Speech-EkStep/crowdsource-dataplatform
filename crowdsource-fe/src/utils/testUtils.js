import { render } from '@testing-library/react';

const Wrapper = ({ children }) => children;

const customRender = (ui, options) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };
