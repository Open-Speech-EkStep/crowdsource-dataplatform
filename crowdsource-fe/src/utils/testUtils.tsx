/* eslint-disable jest/no-export */
import type { ReactNode, ReactElement, ComponentType } from 'react';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render, act } from '@testing-library/react';
import { axe } from 'jest-axe';
import puppeteer from 'puppeteer';
import type { Browser } from 'puppeteer';

const Wrapper: ComponentType = ({ children }: { children?: ReactNode }) => children as ReactElement;

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { renderHook } from '@testing-library/react-hooks';

// override render method
export { customRender as render };

export const verifyAxeTest = ({ container }: { container: RenderResult['container'] }) => {
  it('should not fail an axe audit', async () => {
    let axeResult;

    await act(async () => {
      axeResult = await axe(container);
    });

    expect(axeResult).toHaveNoViolations();
  });
};

export const verifyVRTest = (title: string, url: string) => {
  describe(`${title} VR test`, () => {
    let browser: Browser;

    beforeAll(async () => {
      browser = await puppeteer.launch({
        headless: true,
        devtools: false,
        ignoreHTTPSErrors: true,
        args: ['--start-fullscreen'],
      });
    });

    it('match snapshot', async () => {
      const page = await browser.newPage();
      await page.setViewport({ width: 1920, height: 1080 });

      await page.goto(url, {
        waitUntil: 'networkidle2',
      });

      await page.addStyleTag({
        content: `
      *,
      *::after,
      *::before {
          transition-delay: 0s !important;
          transition-duration: 0s !important;
          animation-delay: -0.0001s !important;
          animation-duration: 0s !important;
          animation-play-state: paused !important;
          caret-color: transparent !important;
      }
      `,
      });

      const image = await page.screenshot({
        fullPage: true,
      });

      expect(image).toMatchImageSnapshot();
    });

    afterAll(async () => {
      await browser.close();
    });
  });
};
