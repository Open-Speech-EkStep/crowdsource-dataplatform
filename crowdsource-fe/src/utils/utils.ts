import { jsPDF } from 'jspdf';
import { i18n } from 'next-i18next';

import { ErrorStatusCode } from 'constants/errorStatusCode';
import { INITIATIVES_MAPPING } from 'constants/initiativeConstants';
import { KEYBOARD_ERROR, LANGUAGE_UNICODE, OTHER_LANGUAGE_UNICODE } from 'constants/Keyboard';
import type { InitiativeType } from 'types/InitiativeType';

import apiPaths from '../constants/apiPaths';

var platform = require('platform');

export const convertIntoHrsFormat = (data: any, isSeconds = true) => {
  const hours = Math.floor(data / 3600);
  const remainingAfterHours = data % 3600;
  const minutes = Math.floor(remainingAfterHours / 60);
  const seconds = Math.round(remainingAfterHours % 60);
  if (isSeconds) {
    return { hours, minutes, seconds };
  } else {
    return { hours, minutes };
  }
};

export const roundOffValue = (average: any, precision: number) => {
  const formattedAverage = average.toFixed(precision);
  return formattedAverage;
};

export const capitalizeFirstLetter = (strValue: string) => {
  return strValue?.charAt(0).toUpperCase() + strValue?.slice(1);
};

const translateText = (text: string) => i18n?.t(text);

export const formatTime = (hours: number, minutes = 0, seconds = 0) => {
  const hrStr = translateText('hours1');
  const minStr = translateText('minutes1');
  const secStr = translateText('seconds1');
  let result = '';
  if (hours > 0) {
    result += `${hours} ${hrStr} `;
  }
  if (minutes > 0) {
    result += `${minutes} ${minStr} `;
  }
  if (hours === 0 && minutes === 0 && seconds > 0) {
    result += `${seconds} ${secStr} `;
  }

  if (hours === 0 && minutes === 0 && seconds === 0) {
    result += `0 ${secStr} `;
  }

  return result.substr(0, result.length - 1);
};

export const convertTimeFormat = (value: any) => {
  const { hours, minutes, seconds } = convertIntoHrsFormat(Number(value) * 60 * 60);
  const data = formatTime(hours, minutes, seconds);
  return data;
};

const roundDuration = (value: number) => {
  return Math.round(value * 100) / 100;
};

export const getHoursValue = (value: any) => {
  return roundDuration(value);
};

export const getMinutesValue = (value: any) => {
  return roundDuration(value * 60);
};

export const getHoursText = (value: any) => {
  return `${getHoursValue(value)} ${i18n?.t('hours2')}`;
};

export const getMinutesText = (value: any) => {
  return `${getMinutesValue(value)} ${i18n?.t('minutes2')}`;
};

export const isSunoOrBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['asr', 'text'];
  return boloOrSuno.includes(value);
};

export const isBoloInitiative = (value: InitiativeType) => {
  const boloOrSuno = ['text'];
  return boloOrSuno.includes(value);
};

export const findInputError = (text: string, currentModule: string, language: string) => {
  if (!text.length) return;
  const newText = text.replace(/\s/g, ''); //read input value, and remove "space" by replace \s
  let error: any = KEYBOARD_ERROR.language;
  const isSunoModule = currentModule === INITIATIVES_MAPPING.suno;
  const specialSymbols = /[\u0021-\u002F\u003A-\u0040\u005B-\u0060\u007B-\u007E\u0964-\u0965]/;
  if (currentModule === INITIATIVES_MAPPING.suno && specialSymbols.test(newText) === true) {
    return KEYBOARD_ERROR.symbol;
  }
  Object.entries(isSunoModule ? LANGUAGE_UNICODE : OTHER_LANGUAGE_UNICODE).forEach(([key, value]) => {
    if (value.test(newText)) {
      //Check Unicode to see which one is true
      if (key.toLowerCase() === language.toLowerCase()) {
        error = KEYBOARD_ERROR.noError;
      }
    }
  });
  return error;
};

export const fetchLocationInfo = async () => {
  const result = await fetch('https://www.cloudflare.com/cdn-cgi/trace')
    .then(res => res.text())
    .then(async ipAddressText => {
      const dataArray = ipAddressText.split('\n');
      let ipAddress = '';
      for (let ind in dataArray) {
        if (dataArray[ind].startsWith('ip=')) {
          ipAddress = dataArray[ind].replace('ip=', '');
          break;
        }
      }
      if (ipAddress.length !== 0) {
        const data = await fetch(`${apiPaths.locationInfo}?ip=${ipAddress}`);
        return data && data.json();
      } else {
        return new Promise((resolve, reject) => {
          reject('Ip Address not available');
        });
      }
    });

  return result;
};

export const getDeviceInfo = () => {
  const os = platform.os;
  return os.family + ' ' + os.version;
};

export const getBrowserInfo = () => {
  return platform.name + ' ' + platform.version;
};

export const downloadBadge = (
  locale: string | undefined,
  initiative: string,
  source: string,
  badgeType: string,
  badgeId: string
) => {
  const pdf = new jsPDF();
  const img = new Image();
  img.src = `/images/${locale}/badges/${locale}_${initiative}_${badgeType}_${source}.png`;
  img.crossOrigin = 'Anonymous';
  img.onload = function () {
    pdf.addImage(img, 50, 10, 105, 130);
    pdf.save(`${badgeType}-badge.pdf`);
  };

  pdf.text(`Badge Id : ${badgeId}`, 36, 190);
};

export const groupBy = function (list: any, keyName: any) {
  if (!list || list.length === 0 || !list[0][keyName]) {
    return [];
  }
  return list.reduce(function (pair: any, key: any) {
    (pair[key[keyName]] = pair[key[keyName]] || []).push(key);
    return pair;
  }, {});
};

/* istanbul ignore next */
export const visualize = (visualizer: any, analyser: any) => {
  const canvasCtx = visualizer.getContext('2d');
  const bufferLength = analyser.frequencyBinCount;
  const dataArray = new Uint8Array(bufferLength);
  const WIDTH = visualizer.width;
  const HEIGHT = visualizer.height;
  // TODO do we need to limit the number of time visualize refreshes per second
  // so that it can run on Android processors without causing audio to drop?
  function draw() {
    // this is more efficient than calling with processor.onaudioprocess
    // and sending floatarray with each call...
    requestAnimationFrame(draw);
    analyser.getByteTimeDomainData(dataArray);
    // canvasCtx.fillStyle = 'rgb(255, 255, 255, 0)';
    // canvasCtx.fillRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.clearRect(0, 0, WIDTH, HEIGHT);
    canvasCtx.lineWidth = 2;
    canvasCtx.strokeStyle = 'rgb(0,123,255)';
    canvasCtx.beginPath();
    const sliceWidth = (WIDTH * 1.0) / bufferLength;
    let x = 0;
    for (let i = 0; i < bufferLength; i++) {
      let v = dataArray[i] / 128.0; // uint8
      let y = (v * HEIGHT) / 2; // uint8
      if (i === 0) {
        canvasCtx.moveTo(x, y);
      } else {
        canvasCtx.lineTo(x, y);
      }
      x += sliceWidth;
    }
    canvasCtx.lineTo(visualizer.width, visualizer.height / 2);
    canvasCtx.stroke();
  }

  draw();
};

export const getErrorMsg = (error: any) => {
  if (error && error.status === ErrorStatusCode.TOO_MANY_REQUEST) {
    return 'multipleRequestApiError';
  } else {
    return 'apiFailureError';
  }
};
