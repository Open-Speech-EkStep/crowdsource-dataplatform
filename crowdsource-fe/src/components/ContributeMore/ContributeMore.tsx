import React from 'react';

import { Trans, useTranslation } from 'next-i18next';
import { useRouter } from 'next/router';

import Button from 'components/Button';
import ImageBasePath from 'components/ImageBasePath';
import Link from 'components/Link';
import { INITIATIVES_MAPPING, INITIATIVES_MEDIA_MAPPING } from 'constants/initiativeConstants';
import { LOCALE_LANGUAGES } from 'constants/localesConstants';
import localStorageConstants from 'constants/localStorageConstants';
import { MEDALS } from 'constants/medalConstants';
import nodeConfig from 'constants/nodeConfig';
import { pageSourceConstants } from 'constants/pageRouteConstants';
import routePaths from 'constants/routePaths';
import useLocalStorage from 'hooks/useLocalStorage';
import type { Initiative } from 'types/Initiatives';
import { capitalizeFirstLetter, downloadBadge, isLanguageImageAvailable } from 'utils/utils';

import styles from './ContributeMore.module.scss';

interface ContributeMoreProps {
  initiative: Initiative;
  source: string;
  url: string;
  nextMileStone: number;
  contributionCount: number;
  nextBadgeType: string;
  badges: Array<any> | undefined;
  isTopLanguage?: string;
  badgeType?: string;
}

const badgeArray = [
  {
    id: 1,
    type: 'Bronze',
  },
  {
    id: 2,
    type: 'Silver',
  },
  {
    id: 3,
    type: 'Gold',
  },
  {
    id: 4,
    type: 'Platinum',
  },
];

const ContributeMore = ({
  initiative,
  source,
  nextMileStone,
  url,
  contributionCount,
  nextBadgeType,
  badges,
  isTopLanguage,
  badgeType,
}: ContributeMoreProps) => {
  const { t } = useTranslation();
  const route = useRouter();
  const [contributionLanguage] = useLocalStorage<string>(localStorageConstants.contributionLanguage);
  const currentContributionAlias = LOCALE_LANGUAGES[contributionLanguage ?? ''];

  const language = isLanguageImageAvailable(currentContributionAlias);

  const download = (badgeType: string, badgeId: string) => {
    downloadBadge(language, initiative, source, badgeType, badgeId);
  };

  return (
    <div
      className={`${styles.root} d-flex align-items-md-center flex-column flex-md-row mx-auto justify-content-md-between p-5 py-md-4 px-md-5 bg-light text-center text-md-start`}
    >
      <div className="d-flex flex-column flex-md-row align-items-md-center">
        <div className="d-flex justify-content-center">
          {badges?.map((item, index) => (
            <Button
              variant="normal"
              key={index}
              title={`Download ${item.grade} Badge`}
              onClick={() => download(item.grade.toLowerCase(), item.generated_badge_id)}
              className={`${styles.medal} d-flex mx-2 flex-shrink-0`}
            >
              <ImageBasePath
                src={`/images/${nodeConfig.brand}/${language}/badges/${language}_${
                  INITIATIVES_MAPPING[initiative]
                }_${item.grade.toLowerCase()}_${source}.svg`}
                width="48"
                height="60"
                alt={`${item.grade.toLowerCase()}Download`}
              />
            </Button>
          ))}
        </div>
        {badgeType === 'Platinum' ? (
          <p className={`${styles.text} mt-3 mt-md-0 ms-md-3 display-4`}>
            {t(`${isTopLanguage}${capitalizeFirstLetter(pageSourceConstants[route.asPath])}Text`, {})}
          </p>
        ) : (
          <p className={`${styles.text} mt-3 mt-md-0 ms-md-3 display-4`}>
            <Trans
              i18nKey={`${INITIATIVES_MAPPING[initiative]}${capitalizeFirstLetter(source)}RemainingText`}
              defaults={`${INITIATIVES_MAPPING[initiative]}${capitalizeFirstLetter(source)}RemainingText`}
              values={{
                count: nextMileStone - contributionCount,
                nextBadgeType: capitalizeFirstLetter(t(nextBadgeType.toLowerCase())),
              }}
              components={{ strong: <strong /> }}
            />
          </p>
        )}

        {badges && badges.length !== 4 && (
          <div className={`${styles.disabled} d-flex justify-content-center mt-4 mt-md-0 mb-3 mb-md-0`}>
            {badgeArray.map((item, index) => {
              return !badges ||
                badges.length === 0 ||
                !badges[index] ||
                (badges && badges.length !== 0 && badges[index] && badges[index].grade !== item.type) ? (
                <div
                  key={index}
                  className={`${styles.medal} ${
                    index === badges.length ? styles.upcomingBadge : styles.otherBadges
                  }  d-flex mx-2 flex-shrink-0`}
                >
                  <ImageBasePath
                    src={`/images/${nodeConfig.brand}/${language}/badges/${language}_${
                      INITIATIVES_MAPPING[initiative]
                    }_${item.type.toLowerCase()}_${source}.svg`}
                    width="48"
                    height="60"
                    alt={`Bronze Badge`}
                  />
                </div>
              ) : null;
            })}
          </div>
        )}
        <Link
          href={`${routePaths.badgeDetail}?initiative=${INITIATIVES_MEDIA_MAPPING[initiative]}&language=${contributionLanguage}&source=${source}&badge=${MEDALS[0]}`}
        >
          <a className={`${styles.text} mt-4 mt-md-0 display-5 ms-md-3 flex-shrink-0`}>
            <b>{t('knowMore')}</b>
          </a>
        </Link>
      </div>
      <Link href={url}>
        <Button className="mt-8 mt-md-0 ms-md-3">{t(`${source}More`)}</Button>
      </Link>
    </div>
  );
};

export default ContributeMore;
