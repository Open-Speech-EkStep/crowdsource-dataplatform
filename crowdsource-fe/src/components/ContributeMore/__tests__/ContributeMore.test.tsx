import router from 'next/router';

import { render, verifyAxeTest, userEvent, screen } from 'utils/testUtils';

import ContributeMore from '../ContributeMore';

describe('ContributeMore', () => {
  const setup = (
    initiative: string,
    source: string,
    nextMileStone: number,
    contributionCount: number,
    nextBadgeType: string,
    badgeType: string
  ) => {
    router.asPath = '/sunoIndia/thank-you';
    const badgeData = [
      {
        generated_badge_id: '4e614ac7-11d8-4126-a3ad-0540fbb4430e',
        grade: 'Bronze',
      },
      {
        generated_badge_id: '4e614ac7-11d8-4126-a3ad-0540fbb44323',
        grade: 'Silver',
      },
      {
        generated_badge_id: '4e614ac7-11d8-4126-a3ad-0540fbb44356',
        grade: 'Gold',
      },
      {
        generated_badge_id: '4e614ac7-11d8-4126-a3ad-0540fbb443124',
        grade: 'Platinum',
      },
    ];
    return render(
      <ContributeMore
        initiative={initiative}
        source={source}
        nextMileStone={nextMileStone}
        contributionCount={contributionCount}
        nextBadgeType={nextBadgeType}
        url=""
        pageMediaTypeStr="sentence(s)"
        badges={badgeData}
        isTopLanguage="see"
        badgeType={badgeType}
      />
    );
  };

  verifyAxeTest(setup('suno', 'contribute', 5, 0, 'Bronze', 'Bronze'));

  it('should render the component and matches it against stored snapshot', () => {
    const { asFragment } = setup('suno', 'contribute', 5, 0, 'Bronze', 'Bronze');

    expect(asFragment()).toMatchSnapshot();
  });

  it('after user earn the platinum badge and should download all the badges', () => {
    setup('suno', 'contribute', 200, 0, 'Platinum', 'Platinum');

    userEvent.click(screen.getByRole('img', { name: 'bronzeDownload' }));

    userEvent.click(screen.getByRole('img', { name: 'silverDownload' }));

    userEvent.click(screen.getByRole('img', { name: 'goldDownload' }));

    userEvent.click(screen.getByRole('img', { name: 'platinumDownload' }));

    expect(screen.getByText('seeContributeText')).toBeInTheDocument();
  });
});
