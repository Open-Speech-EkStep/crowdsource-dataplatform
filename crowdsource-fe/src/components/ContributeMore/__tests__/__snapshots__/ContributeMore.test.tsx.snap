// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributeMore should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <div
    class="root d-flex align-items-md-center flex-column flex-md-row mx-auto justify-content-md-between p-5 py-md-4 px-md-5 bg-light text-center text-md-start"
  >
    <div
      class="d-flex flex-column flex-md-row align-items-md-center"
    >
      <div
        class="d-flex justify-content-center"
      >
        <button
          class="medal d-flex mx-2 flex-shrink-0 info normal btn btn-link"
          data-testid="Button"
          title="Download Bronze Badge"
          type="button"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDgiIGhlaWdodD0iNjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="bronzeDownload"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </button>
        <button
          class="medal d-flex mx-2 flex-shrink-0 info normal btn btn-link"
          data-testid="Button"
          title="Download Silver Badge"
          type="button"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDgiIGhlaWdodD0iNjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="silverDownload"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </button>
        <button
          class="medal d-flex mx-2 flex-shrink-0 info normal btn btn-link"
          data-testid="Button"
          title="Download Gold Badge"
          type="button"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDgiIGhlaWdodD0iNjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="goldDownload"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </button>
        <button
          class="medal d-flex mx-2 flex-shrink-0 info normal btn btn-link"
          data-testid="Button"
          title="Download Platinum Badge"
          type="button"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDgiIGhlaWdodD0iNjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="platinumDownload"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </button>
      </div>
      <p
        class="text mt-3 mt-md-0 ms-md-3 display-4"
      >
        ttsContributeRemainingText
      </p>
      <a
        class="text mt-4 mt-md-0 display-5 ms-md-3 flex-shrink-0"
        href="/badges?initiative=asr&language=null&source=contribute&badge=bronze"
      >
        <b>
          knowMore
        </b>
      </a>
    </div>
    <button
      class="mt-8 mt-md-0 ms-md-3 info primary btn btn-primary"
      data-testid="Button"
      type="button"
    >
      contributeMore
    </button>
  </div>
</DocumentFragment>
`;
